mongo/integration/collection_test.go

```diff
@@ -74,6 +74,37 @@ func TestCollection(t *testing.T) {
 			assert.True(mt, ok, "expected error type %v, got %v", mongo.WriteException{}, err)
 			assert.NotNil(mt, we.WriteConcernError, "expected write concern error, got %+v", we)
 		})
+
+		// Require 3.2 servers for bypassDocumentValidation support.
+		nilOptsOpts := mtest.NewOptions().MinServerVersion("3.2")
+		mt.RunOpts("nil opts", nilOptsOpts, func(mt *mtest.T) {
+			assertOptSet := func(expectSet bool) {
```

Comment:
This might be better organized/readable as a set of table-driven tests? If you have this sort of local helper function that gets called for different configurations in the test, it's usually a sign that things can be refactored into a table that runs sub-tests.

Reply:
Oh good idea... I just learned that this was a thing outside of the Go driver from the testing talk in the onboarding doc.